### DISCUSSION TITLE
C# Markup: How to bind TextBlock().Run().Text() to the 'model'?

### DISCUSSION THREAD

**Discussion by kucint** [unoplatform/uno#15198] [Q&A] [answered]
Here is the code, that displays a piece of text using regular and bold fonts:
```
new TextBlock()
.Inlines(
    new Run().Text("Regular text "),
    new Run().Text("Bold Text ")
        .FontWeight(FontWeights.SemiBold)),
```

Let's suppose I want to replace static "Bold Text" with a text coming from the 'model'. 
Unfortunately `Run().Text()` accepts only a string but not a binding delegate ...
I tried the following:
```
new TextBlock()
.Inlines(
    new Run().Text("Regular text "),
    new Run().Text($"{((BindableMainModel)DataContext).TextFromModel}")
        .FontWeight(FontWeights.SemiBold)),
```

This shows the requested text but the text obviously will not be updated when TextFromModel changes.
How am I suppose to do it?

**Comment by dansiegel** [comment, marked as answer]
What you need here is the `MarkupRun`. This [article in the docs](https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Markup/Appendix/TextBlockRun.html) should explain what you're looking for in more detail. Please let us know if there is anything that is unclear. The team is always looking to improve the docs.

**Comment by kucint** [comment]
Great!
I will have a look and let you know about any possible improvements.
Thanks @dansiegel .

**Comment by kucint** [comment]
IMO, Documentation is doesn't need any further improvements.
it is clear and straightforward.
thanks!
